subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,29 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+	/*
+	 * If we receive RST here, we close the socket. See RFC 793 chapter
+	 * called "Reset Processing" for details.
+	 */
+	if (tcp_flags & NET_TCP_RST) {
+		/* We only accept RST packet that has valid seq field. */
+		if (!tcp_validate_seq(context->tcp,
```

Comment:
The second argument to `tcp_validate_seq()` is always the same; couldn't one just pass a pkt and the call to `sys_get_be32(...)` happen inside it? This should save a few bytes in text.